### PR TITLE
build: upgrade tool versions

### LINKED ISSUE #154: build: upgrade tool versions
**Is your feature request related to a problem? Please describe.**

Salmon currently prints the following message to `STDERR`:

```console
Version Info: ### PLEASE UPGRADE SALMON ###
### A newer version of salmon with important bug fixes and improvements is available. ####
###
The newest version, available at https://github.com/COMBINE-lab/salmon/releases
contains new features, improvements, and bug fixes; please upgrade at your
earliest convenience.
###
```

The Salmon version currently used in ZARP is 1.4.0 - the latest available version is 1.10.1.

Furthermore, #152 seems to have arisen spontaneously and seems to be fixed by upgrading Cutadapt (#153).

These observations highlight that the versions of the tools we use in ZARP are becoming outdated. It might be good to upgrade the tools to the latest versions available on Bioconda before publishing the ZARP manuscript.

**Describe the solution you'd like**

Upgrade the (Bio)Conda and Docker image versions of all tools to a more recent minor version. Do not upgrade major versions, unless it was verified that any changes to a tool's API do not have an impact on ZARP wiring.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by uniqueg**
## Description

- Upgraded most tools to latest versions in Conda environments in `workflow/envs`
- Upgraded Docker containers accordingly
- Upgrade Ubuntu Docker images from Focal Fossa to Jammy Jellyfish
- Use `sra-tools` and `pigz` images from BioContainers (now all images except Ubuntu-based ones are from BioContainers)

Fixes #154 

## Type of change

- [x] Maintenance

## Conventional Commits guidelines

- [x] I made sure the PR title follows the 
https://www.conventionalcommits.org/en/v1.0.0/

## Checklist:

- [x] My code changes follow the style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] I have updated the project's documentation

### CODE REVIEW

**Comment by uniqueg** [comment]
Expecting this to fail, of course - at the very least the expected outputs have to be updated. The main questions for now are (a) whether the tests run through at all and (b) whether the results are compatible.

**Comment by uniqueg** [comment]
In all tests, we currently get an error in rule `create_kallisto_index`. Will restore the old version of kallisto and see if the error goes away

**Comment by uniqueg** [comment]
The problem with kallisto index likely stems from this release that introduced a new and improved kallisto index:
https://github.com/pachterlab/kallisto/releases/tag/v0.50.0

I have now set the version to the last version before the offending version (`0.50.0`), so now it's `0.48.0`, up from `0.46.2` that we previously used. With that all test workflows run through. As expected, the tests still fail, as many of the outputs changed (different MD5 sums).

Now, the question is whether they change in a way that we believe is _worse_ than before. Do we have any way of assessing the testing quantitatively? How have we previously done this when upgrading tools (I'm sure this is not the first time)? Just ran a pipeline on one or more known samples and checked if the results look roughly the same and, if so, just upgraded? Because I think I would like to do that here as well. What do you think @mkatsanto?

@balajtimate As discussed, I rolled back the changes to the HTSinfer Docker image from this PR, so that it'll still pull the latest image from the Zavolab registry after merging. We can put in a separate PR for that when the image is fixed on BioContainers.

**Comment by uniqueg** [comment]
Actually, looking at the 57 files whose MD5 sum changed, it's not too bad at all:
- 56 are related to FastQC, so they have no downstream dependencies, e.g., on alignments, quantification etc.
- 1 is a file with version info (expected to change)

So I think the question about still getting compatible outputs doesn't arise.

Here's a list of the files whose MD5 check failed (renamed and grouped to make them easier to read/process):

```console
fastqc/fq1/single/fastqc_data.txt
fastqc/fq1/single/fastqc.fo
fastqc/fq1/single/Images/adapter_content.png
fastqc/fq1/single/Images/duplication_levels.png
fastqc/fq1/single/Images/per_base_n_content.png
fastqc/fq1/single/Images/per_base_sequence_content.png
fastqc/fq1/single/Images/per_sequence_gc_content.png
fastqc/fq1/single/Images/per_sequence_quality.png
fastqc/fq1/single/Images/sequence_length_distribution.png
fastqc/fq1/paired/fastqc_data.txt
fastqc/fq1/paired/fastqc.fo
fastqc/fq1/paired/Images/adapter_content.png
fastqc/fq1/paired/Images/duplication_levels.png
fastqc/fq1/paired/Images/per_base_n_content.png
fastqc/fq1/paired/Images/per_base_sequence_content.png
fastqc/fq1/paired/Images/per_sequence_gc_content.png
fastqc/fq1/paired/Images/per_sequence_quality.png
fastqc/fq1/paired/Images/sequence_length_distribution.png
fastqc/fq2/paired/fastqc_data.txt
fastqc/fq2/paired/fastqc.fo
fastqc/fq2/paired/Images/adapter_content.png
fastqc/fq2/paired/Images/duplication_levels.png
fastqc/fq2/paired/Images/per_base_n_content.png
fastqc/fq2/paired/Images/per_base_sequence_content.png
fastqc/fq2/paired/Images/per_sequence_gc_content.png
fastqc/fq2/paired/Images/per_sequence_quality.png
fastqc/fq2/paired/Images/sequence_length_distribution.png
fastqc_trimmed/fq1/single/fastqc_data.txt
fastqc_trimmed/fq1/single/fastqc.fo
fastqc_trimmed/fq1/single/Images/adapter_content.png
fastqc_trimmed/fq1/single/Images/duplication_levels.png
fastqc_trimmed/fq1/single/Images/per_base_n_content.png
fastqc_trimmed/fq1/single/Images/per_base_sequence_content.png
fastqc_trimmed/fq1/single/Images/per_sequence_gc_content.png
fastqc_trimmed/fq1/single/Images/per_sequence_quality.png
fastqc_trimmed/fq1/single/Images/sequence_length_distribution.png
fastqc_trimmed/fq1/paired/fastqc_data.txt
fastqc_trimmed/fq1/paired/fastqc.fo
fastqc_trimmed/fq1/paired/Images/adapter_content.png
fastqc_trimmed/fq1/paired/Images/duplication_levels.png
fastqc_trimmed/fq1/paired/Images/per_base_n_content.png
fastqc_trimmed/fq1/paired/Images/per_base_sequence_content.png
fastqc_trimmed/fq1/paired/Images/per_sequence_gc_content.png
fastqc_trimmed/fq1/paired/Images/per_sequence_quality.png
fastqc_trimmed/fq1/paired/Images/sequence_length_distribution.png
fastqc_trimmed/fq2/paired/fastqc_data.txt
fastqc_trimmed/fq2/paired/fastqc.fo
fastqc_trimmed/fq2/paired/Images/adapter_content.png
fastqc_trimmed/fq2/paired/Images/duplication_levels.png
fastqc_trimmed/fq2/paired/Images/per_base_n_content.png
fastqc_trimmed/fq2/paired/Images/per_base_sequence_content.png
fastqc_trimmed/fq2/paired/Images/per_sequence_gc_content.png
fastqc_trimmed/fq2/paired/Images/per_sequence_quality.png
fastqc_trimmed/fq2/paired/Images/sequence_length_distribution.png
multiqc_summary/multiqc_data/multiqc_fastqc_fastqc_raw.txt
multiqc_summary/multiqc_data/multiqc_fastqc_fastqc_trimmed.txt
salmon_indexes/homo_sapiens/31/salmon.idx/versionInfo.json
```

File: tests/test_format/test_snakefmt.sh
Comment (mkatsanto) [inline tests/test_format/test_snakefmt.sh:1]: What changed here? I only see an empty description.

File: tests/test_format/test_snakemake_lint.sh
Comment (mkatsanto) [inline tests/test_format/test_snakemake_lint.sh:1]: What changed here? I only see an empty description.

File: tests/test_format/test_snakefmt.sh
Comment (uniqueg) [reply tests/test_format/test_snakefmt.sh:1]: I made all scripts executable, for consistency. Before, some scripts had the +x flag, others didn't.

**Review by mkatsanto — APPROVED** [review]

**Pull request merged**